PaymentHighway/Core/CardBrand.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  PHCardType.swift
+//  PaymentHighway
+//
+//  Copyright © 2018 Payment Highway Oy. All rights reserved.
+//
+
+import Foundation
+
+/// Card brands to which a payment card can belong.
+///
+public enum CardBrand {
+    
+    /// Visa card
+    case visa
+    
+    /// Mastercard card
+    case mastercard
+    
+    /// American Express card
+    case americanExpress
+    
+    /// Discover card
+    case discover
+    
+    /// JCB card
+    case jcb
+    
+    /// Dinners Club card
+    case dinersClub
+    
+    /// return array with all card brands
+    public static var allCases: [CardBrand] {
+        return [.visa, .mastercard, .americanExpress, .discover, .jcb, .dinersClub]
+    }
+
+}
+
+// swiftlint:disable force_try
+extension CardBrand {
+    
+    /// Returns the pattern to recognise a card brand
+    fileprivate var pattern: NSRegularExpression {
+        var regexp: String
+        switch self {
+        case .visa:
+            regexp = "^4[0-9]"
+        case .mastercard:
+            regexp = "^(?:5[1-5]|2[2-7])"
+        case .americanExpress:
+            regexp = "^3[47]"
+        case .discover:
+            regexp = "^(?:6[045]|622)"
+        case .jcb:
+            regexp = "^35"
+        case .dinersClub:
+            regexp = "^3[0689]"
+        }
+        return try! NSRegularExpression(pattern: regexp)
+    }
+    
+    /// Recognize the card brand of a credit card number
+    ///
+    /// - parameter cardNumber: The card number string
+    static func from(cardNumber: String) -> CardBrand? {
```

Comment:
Should this be an `init?` method instead?

Reply:
Agree. fixed. Thanks!